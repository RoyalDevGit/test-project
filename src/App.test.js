import axios from 'axios';
import { render, screen } from '@testing-library/react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from './App';

jest.mock('axios');
configure({ adapter: new Adapter() });

// test the api 
describe('mock api call', () => {
  test("get data from axios api call", () => {
    const mockedResponse = { data: { name: "Auxiliary Generating Microchip" } };
    axios.get.mockResolvedValue(mockedResponse);

    const app = require('./layout/Layout');
    app.apiCall();
    expect(axios.get).toHaveBeenCalled();
    expect(axios.get).toHaveBeenCalledWith('https://idme-interview.herokuapp.com');
  })
})

describe('rendering on the different viewports', () => {
  // test on the laptop
  it('renders correctly on the laptop', () => {
    window.innerWidth = 1200;
    render(<App />);
    expect(screen.queryByTestId('laptop-view')).toHaveStyle({ display: 'table' });
    expect(screen.queryByTestId('mobile-view')).toHaveStyle({ display: 'none' });
  })

  // test on the mobile
  it('renders correctly on the mobile', () => {
    window.innerWidth = 500;
    render(<App />);
    expect(screen.queryByTestId('laptop-view')).toHaveStyle({ display: 'table' });
    expect(screen.queryByTestId('mobile-view')).toHaveStyle({ display: 'block' });
  })
})






