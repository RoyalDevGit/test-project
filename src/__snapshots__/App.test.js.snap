// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering on the different mobile renders correctly on the mobile 1`] = `
<section
  className="container"
>
  <div
    className="title"
    id="Purchases"
  >
    Purchases
  </div>
  <table
    className="table"
  >
    <thead>
      <tr
        className="header"
      >
        <th
          className="margin-cell"
        />
        <th
          className="table-header-cell"
        >
          Name
        </th>
        <th
          className="table-header-cell"
        >
          Location
        </th>
        <th
          className="table-header-cell"
        >
          Purchase Date
        </th>
        <th
          className="table-header-cell category-header"
        >
          Category
        </th>
        <th
          className="table-header-cell"
        >
          Description
        </th>
        <th
          className="table-header-cell"
        >
          Price
        </th>
        <th
          className="margin-cell"
        />
      </tr>
    </thead>
    <tbody
      className="table-body"
    />
  </table>
  <div
    className="mobile-device"
  />
</section>
`;

exports[`rendering on the different viewports renders correctly on the laptop 1`] = `
<section
  className="container"
>
  <div
    className="title"
    id="Purchases"
  >
    Purchases
  </div>
  <table
    className="table"
  >
    <thead>
      <tr
        className="header"
      >
        <th
          className="margin-cell"
        />
        <th
          className="table-header-cell"
        >
          Name
        </th>
        <th
          className="table-header-cell"
        >
          Location
        </th>
        <th
          className="table-header-cell"
        >
          Purchase Date
        </th>
        <th
          className="table-header-cell category-header"
        >
          Category
        </th>
        <th
          className="table-header-cell"
        >
          Description
        </th>
        <th
          className="table-header-cell"
        >
          Price
        </th>
        <th
          className="margin-cell"
        />
      </tr>
    </thead>
    <tbody
      className="table-body"
    />
  </table>
  <div
    className="mobile-device"
  />
</section>
`;

exports[`rendering on the different viewports renders correctly on the mobile 1`] = `
<section
  className="container"
>
  <div
    className="title"
    id="Purchases"
  >
    Purchases
  </div>
  <table
    className="table"
  >
    <thead>
      <tr
        className="header"
      >
        <th
          className="margin-cell"
        />
        <th
          className="table-header-cell"
        >
          Name
        </th>
        <th
          className="table-header-cell"
        >
          Location
        </th>
        <th
          className="table-header-cell"
        >
          Purchase Date
        </th>
        <th
          className="table-header-cell category-header"
        >
          Category
        </th>
        <th
          className="table-header-cell"
        >
          Description
        </th>
        <th
          className="table-header-cell"
        >
          Price
        </th>
        <th
          className="margin-cell"
        />
      </tr>
    </thead>
    <tbody
      className="table-body"
    />
  </table>
  <div
    className="mobile-device"
  />
</section>
`;

exports[`renders correctly 1`] = `
<section
  className="container"
>
  <div
    className="title"
  >
    Purchases
  </div>
  <table
    className="table"
  >
    <thead>
      <tr
        className="header"
      >
        <th
          className="margin-cell"
        />
        <th
          className="table-header-cell"
        >
          Name
        </th>
        <th
          className="table-header-cell"
        >
          Location
        </th>
        <th
          className="table-header-cell"
        >
          Purchase Date
        </th>
        <th
          className="table-header-cell category-header"
        >
          Category
        </th>
        <th
          className="table-header-cell"
        >
          Description
        </th>
        <th
          className="table-header-cell"
        >
          Price
        </th>
        <th
          className="margin-cell"
        />
      </tr>
    </thead>
    <tbody
      className="table-body"
    />
  </table>
  <div
    className="mobile-device"
  />
</section>
`;
